### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by alicerdv**
## What is the goal of this PR?

Update doc strings and update sphinx documentation to reflect the new changes we have recently introduced such as the addition of the Pipeline and Oracle

## What are the changes implemented in this PR?

Update the sphinx conf.py file so that
1) Doc strings from __init__ are rendered (`autoclass_content = "both"`)
2) The path of the class is not displayed, only the name of the class (`add_module_names = False`)

Update the intro and examples in the sphinx documentation to reflect the new changes we have recently made
Update the sphinx documentation so that the pipeline and oracle modules are rendered
Update missing doc strings

### CODE REVIEW

**Comment by alicerdv** [comment]
This isn't finished yet, it will need to be updated after any name change PRs are merged

**Comment by alicerdv** [comment]
Some of the images/code examples might need to be updated to align with the paper. I wasn't sure which images we wanted to keep

**Review by a-pouplin — DISMISSED** [review]
Nice ! Everything seems good on my side :)

File: docs/source/notes/quick_start.rst
Comment (thomasgaudelet) [inline docs/source/notes/quick_start.rst:9]: we should add a mention here that the pyrelational datamanager makes some assumptions about dataset. e.g

- it assumes that the labels are contained in an independent attribute of the dataset (unlike TensorDatasets) that can be specified with the `label_attr` input to the data manager
- that the dataset __getitem __ return a tuple of tensors
- a number of methods assumes the featurization is entirely contained in a single tensor, corresponding to the first element of the tuple returned by the __getitem __

Comment (thomasgaudelet) [reply docs/source/notes/quick_start.rst:9]: maybe to add to the readme too

File: docs/Makefile
Comment (paulmorio) [inline docs/Makefile:8]: Oh how I wish we were able to get that :D. Thanks for spotting this.

File: pyrelational/model_managers/abstract_model_manager.py
Comment (paulmorio) [inline pyrelational/model_managers/abstract_model_manager.py:13]: I think this needs to be updated to `ActiveLearningStrategy` -> `Strategy`

File: pyrelational/model_managers/ensemble_model_manager.py
Comment (paulmorio) [inline pyrelational/model_managers/ensemble_model_manager.py:47]: Thanks for adding this detail!

File: docs/source/reference/strategies.rst
Comment (paulmorio) [inline docs/source/reference/strategies.rst:20]: This is good

File: pyrelational/strategies/regression/bald_strategy.py
Comment (paulmorio) [inline pyrelational/strategies/regression/bald_strategy.py:34]: I had to look up if peakier is a proper description. In some circles it is! I learned something today.

**Review by paulmorio — CHANGES_REQUESTED** [review]
Thanks for the hard work and all the clarifications on the method, its a huge improvement for the library. I just added one comment where I saw some reference to a deprecated module. Should approve after.

File: docs/source/notes/using_your_own_strategy.rst
Comment (thomasgaudelet) [inline docs/source/notes/using_your_own_strategy.rst:76]: train_and_infer might be useful here?

File: pyrelational/strategies/regression/bald_strategy.py
Comment (thomasgaudelet) [reply pyrelational/strategies/regression/bald_strategy.py:34]: It is a weird word one for sure

File: pyrelational/model_managers/abstract_model_manager.py
Comment (alicerdv) [reply pyrelational/model_managers/abstract_model_manager.py]: thanks, I have changed it now :)

File: docs/source/notes/using_your_own_strategy.rst
Comment (alicerdv) [reply docs/source/notes/using_your_own_strategy.rst]: I've updated it now

File: docs/source/notes/quick_start.rst
Comment (alicerdv) [reply docs/source/notes/quick_start.rst]: good points, I have added it here, in the doc string and in the readme

**Review by thomasgaudelet — APPROVED** [review]

**Review by paulmorio — APPROVED** [review]
Approved

**Pull request merged**
